src/extension/extension.ts

```diff
@@ -24,186 +44,205 @@ export function activate(context: vscode.ExtensionContext): void {
       solc: solcPath + solc,
       staging: branch,
     }
-
-    if (solcArgs) {
-      confFileDefault['--solc-args'] = [solcArgs]
+    if (solcArgs !== '{}') {
+      confFileDefault.solc_args = ''
+      const tempArgs = JSON.parse(solcArgs)
+      Object.entries(tempArgs).forEach(arg => {
+        let tempValue = arg[1]
+        if (tempValue) {
+          tempValue += ','
+        }
+        confFileDefault.solc_args +=
+          '--' + arg[0].replace('--', '') + ',' + tempValue
+      })
+      confFileDefault.solc_args = confFileDefault.solc_args
+        .toString()
+        .replace(/.$/, '')
+        .split(',')
     }
     if (defaultDirectoryForPackagesDependencies) {
-      confFileDefault['--packages_path'] =
-        defaultDirectoryForPackagesDependencies
+      confFileDefault.packages_path = defaultDirectoryForPackagesDependencies
     }
     if (solidityPackageDirectories !== '{}') {
-      confFileDefault['--packages'] = solidityPackageDirectories
+      confFileDefault.packages = solidityPackageDirectories
+        .replace('{', '')
+        .replace('}', '')
+        .split(',')
     }
     if (optimisticLoop) {
-      confFileDefault['--optimistic_loop'] = true
+      confFileDefault.optimistic_loop = true
     }
-    if (loopUnroll !== 1) {
-      confFileDefault['--loop_iter'] = loopUnroll
+    if (loopUnroll) {
+      confFileDefault.loop_iter = loopUnroll
     }
-    if (duration !== 600) {
-      confFileDefault['--smt_timeout'] = duration
+    if (duration) {
+      confFileDefault.smt_timeout = duration
     }
-    if (additionalSettings !== '{}') {
-      const settingsArray: string[] = []
-      Object.entries(JSON.parse(additionalSettings)).forEach(([key, value]) => {
-        if (!key.startsWith('-')) {
-          key = '-' + key
-        }
-        const setting = key + (value ? '=' + value : '')
-        settingsArray.push(setting)
+    if (additionalFlags !== '{}') {
+      Object.entries(JSON.parse(additionalFlags)).forEach(([key, value]) => {
+        confFileDefault[key.toString()] = value as string
       })
-      confFileDefault['--settings'] = settingsArray
     }
-
     if (!typeCheck) {
-      confFileDefault['--typecheck_only'] = true
+      confFileDefault.disableLocalTypeChecking = true
+    } else if (typeCheck === true) {
+      confFileDefault.disableLocalTypeChecking = false
     }
-
     return confFileDefault
   }
 
-  async function quickPickWithConfFiles(
-    select: (
-      selection: readonly vscode.QuickPickItem[],
-      quickPick: vscode.QuickPick<vscode.QuickPickItem>,
-      basePath: vscode.WorkspaceFolder,
-    ) => void,
-  ) {
+  /**
+   * open the settings webview to edit the content of a conf file
+   * @param name the name of a run to edit, in a type that holds both conf file name
+   * and display name of the run
+   * @returns Promise<void>
+   */
+  async function editConf(name: ConfNameMap): Promise<void> {
     const path = vscode.workspace.workspaceFolders?.[0]
```

Comment:
reused code:
devide to functions: creating uri, reading file

Reply:
* used getConfUri where there was same code
* added readConf, renderSettingsView functions